### PR TITLE
Implement IMethodSymbol.ReceiverNullableAnnotation

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/35034. @dotnet/roslyn-compiler please review.

### CODE REVIEW

**Comment by 333fred** [comment]
Ping @dotnet/roslyn-compiler for a review.

File: src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs [hunk 1]
```diff
@@ -1097,8 +1097,10 @@ ITypeSymbol IMethodSymbol.ReceiverType
             }
         }
 
-        // https://github.com/dotnet/roslyn/issues/35034: Implement
-        CodeAnalysis.NullableAnnotation IMethodSymbol.ReceiverNullableAnnotation => default;
+        CodeAnalysis.NullableAnnotation IMethodSymbol.ReceiverNullableAnnotation => ReceiverNullableAnnotation;
+
+        protected virtual CodeAnalysis.NullableAnnotation ReceiverNullableAnnotation =>
+            IsStatic ? CodeAnalysis.NullableAnnotation.NotApplicable : CodeAnalysis.NullableAnnotation.NotAnnotated;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs]: >CodeAnalysis.NullableAnnotation.NotAnnotated [](start = 71, length = 44)

Should we always return ```NotApplicable``` unless the method is a reduced extension method?

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs [hunk 2]
```diff
@@ -358,7 +358,7 @@ public class C
                 PublicNullableAnnotation.Annotated);
         }
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/35034")]
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs]: >https://github.com/dotnet/roslyn/issues/35034 [](start = 22, length = 45)

Consider capturing the link in WorkItem attribute.

File: src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs]: @cston and I debated this a bit yesterday afternoon, and I ended up deciding that it felt more correct to have it be `NotAnnotated` unless the method didn't have a receiver. We warn when you invoke an instance method on a not null receiver, and it feels like the public API should reflect this fact.

---
In reply to: [277793575](https://github.com/dotnet/roslyn/pull/35191#discussion_r277793575) [](ancestors = 277793575)

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs (hunk 2 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs]: I'll address this in a follow up PR.

---
In reply to: [277794882](https://github.com/dotnet/roslyn/pull/35191#discussion_r277794882) [](ancestors = 277794882)

**Pull request merged**